### PR TITLE
add chicago_taxi data set

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR adds the chicago data set. I chose to use the 2022 data set as it is static. So if we need to pull again we should have fewer issues.

For use in some workshops. The data right now is preprocessed in a way such that minimal {recipes} work has to be done to be effective.

some not needed variables are included for discussion but should properly be removed doing modeling.

I talked to Max about the data, and we had the idea of removing cash payments as they are likely reported incorrectly

```r
# A tibble: 6 × 3
cash  no_tips       n
<lgl> <lgl>     <int>
FALSE FALSE   3220346
FALSE TRUE    1088281
TRUE  FALSE     10750
TRUE  TRUE    2059512
```

`glimpse()` of the data below:

```r
Rows: 10,000
Columns: 19
$ tip                    <fct> yes, yes, yes, no, yes, yes, yes, yes, yes, yes, yes,…
$ trip_id                <fct> 22dfccfa337d3ef59d99ed3affaa0949297b0d66, 8ea34d47f54…
$ trip_seconds           <dbl> 2160, 2168, 600, 3120, 2423, 1688, 780, 520, 494, 327…
$ trip_miles             <dbl> 12.80, 17.30, 1.30, 7.70, 19.59, 16.53, 0.20, 1.52, 1…
$ pickup_community_area  <fct> 56, 76, 8, 69, 56, 76, 8, 8, 7, NA, 8, 8, 7, 76, 5, 2…
$ dropoff_community_area <fct> 8, 8, 32, 43, NA, 8, 28, 32, 8, 32, 24, 25, 8, 76, NA…
$ fare                   <dbl> 36.00, 42.50, 7.50, 32.00, 50.75, 41.25, 14.50, 9.89,…
$ tips                   <dbl> 8.05, 5.00, 4.00, 0.00, 10.00, 11.44, 2.00, 2.27, 0.5…
$ tolls                  <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,…
$ extras                 <dbl> 4.0, 5.5, 0.0, 0.0, 32.0, 4.0, 0.0, 0.0, 0.0, 4.0, 0.…
$ trip_total             <dbl> 48.05, 53.50, 11.50, 32.00, 93.25, 57.19, 16.50, 12.1…
$ payment_type           <fct> Credit Card, Credit Card, Credit Card, Unknown, Credi…
$ company                <fct> "Taxi Affiliation Services", "Globe Taxi", "Top Cab A…
$ trip_start             <dttm> 2022-06-19 20:15:00, 2022-02-14 15:00:00, 2022-04-22…
$ local_trip             <fct> no, no, no, no, NA, no, no, no, no, NA, no, no, no, y…
$ trip_start_dow         <fct> Sun, Mon, Fri, Sat, Wed, Tue, Fri, Sat, Fri, Sun, Sun…
$ trip_start_month       <fct> Jun, Feb, Apr, Apr, Nov, Feb, May, Feb, Aug, Jul, Sep…
$ trip_start_hour        <int> 20, 15, 18, 11, 16, 18, 0, 9, 23, 12, 12, 0, 20, 10, …
$ trip_start_minute      <int> 15, 0, 0, 45, 45, 45, 15, 45, 0, 30, 15, 15, 45, 45, …
```

### CODE REVIEW

**Comment by EmilHvitfeldt** [comment]
You can play around with this data set by calling 

```r
chicago_taxi <- readr::read_rds("https://github.com/tidymodels/modeldatatoo/raw/taxi/pkgdown/assets/pins-board/chicago_taxi/20230627T221924Z-598e2/chicago_taxi.rds")
```

File: data-raw/chicago_taxi.R [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## code to prepare `chicago_taxi` dataset goes here
+
+library(tidyverse)
+library(tidymodels)
+library(janitor)
+
+# https://data.cityofchicago.org/Transportation/Taxi-Trips-2022/npd7-ywjz
+taxi_raw <- read_csv("~/Downloads/Taxi_Trips_-_2022.csv") |>
```
Comment (topepo) [inline data-raw/chicago_taxi.R:8]: I think that you can get a url for the api. That's what we did with the train data. [This url](https://data.cityofchicago.org/api/views/e55j-2ewb/rows.csv?accessType=DOWNLOAD) should work (and you can probably get one with parameters too).

**Review by simonpcouch — COMMENTED** [review]
Thank you for making this happen! 

re:

> some not needed variables are included for discussion but should properly be removed doing modeling.

I think this is a good move and I'm on board. The discussions for why some of these variables ought not to be included are longer than others, and I think altogether we may be spending a good while talking about these if we don't pare a couple more back.

It's my sense that the variables that wouldn't be part of that logistic regression are `pickup_community_area`, `dropoff_community_area` (too many levels -> rank-deficiency, information encoded in `local_trip`), `tips` (proxy for the outcome), `trip_start`, `trip_start_minute` (since rounded so coarsely), `company` (too many levels). Perhaps get rid of `*_community_area`, maybe in favor of derived variables on demographic information if we wanted to include that information in the model? Maybe we could `step_other(company)`?

```
> table(chicago_taxi$company)

             2733 - 74600 Benny Jona        312 Medallion Management Corp 
                                   5                                   14 
         3556 - 36214 RC Andrews Cab             4787 - 56058 Reny Cab Co 
                                   2                                    1 
                         5 Star Taxi           5167 - 71969 5167 Taxi Inc 
                                 453                                    2 
         6574 - Babylon Express Inc.         Blue Ribbon Taxi Association 
                                   1                                  297 
                Chicago Independents                      Chicago Taxicab 
                                 616                                   30 
             Choice Taxi Association          Choice Taxi Association Inc 
                                 154                                   42 
                        City Service                            Flash Cab 
                                1095                                 1827 
                          Globe Taxi                Koam Taxi Association 
                                 372                                   24 
                    Medallion Leasin                    Metro Jet Taxi A. 
                                 362                                    1 
Patriot Taxi Dba Peace Taxi Associat                           Setare Inc 
                                  23                                    7 
      Star North Taxi Management Llc                             Sun Taxi 
                                 289                                 1131 
           Taxi Affiliation Services Taxi Affiliation Services Llc - Yell 
                                1803                                    8 
        Taxicab Insurance Agency Llc        Taxicab Insurance Agency, LLC 
                                1035                                  201 
                             Top Cab                  Top Cab Affiliation 
                                  20                                  129 
                           U Taxicab 
                                  56 
```

**Comment by hfrick** [comment]
I'd be okay with the discussion going along the lines of: for logistic regression, you wouldn't include these `<decide_me>` variables because of how the model works. If you want them in the model, you can switch, e.g., to a random forest or do preprocessing. And we let participants take the random forest route and keep preprocessing in our back pocket. (And we can talk about the temptation to do preprocessing with dplyr on the entire data instead of the resamples in the introduction to workflows.)

File: data-raw/chicago_taxi.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply data-raw/chicago_taxi.R]: [`0275faa` (#10)](https://github.com/tidymodels/modeldatatoo/pull/10/commits/0275faa5a9aa948c40747c023e9aa8390e5feb1b)

**Comment by EmilHvitfeldt** [comment]
Alright, I did a couple of things.

Removed
- tips
- trip_start
- trip_start_minute
- both community_area ones

Problematic ones still left:
- trip_seconds
- trip_total, (since we can calculate `tips` as `trip_total - fare - tolls - extras`
- trip_id
- company (I used `step_other()` on it, so we need to talk about why we did that)

I'm gonna merge it now to keep the momentum. We can change it if needs be

**Pull request merged**